src/main/java/edu/kit/compiler/assembly/ElfAssemblyWriter.java

```diff
@@ -0,0 +1,143 @@
+
+/**
+ * Writes the assembly file in the ELF format.
+ */
+public class ElfAssemblyWriter implements AssemblyWriter {
+
+    private static final int DEFAULT_COMMAND_INDENT = 8;
+    private static final int DEFAULT_COMMENT_INDENT = 40;
+    private static final String COMMENT_START_CHAR = "#";
+    private static final String FUNCTION_ALIGNMENT = "4,,15";
+
+    private PrintStream output;
+
+    private String format(String format, Object... args) {
+        return String.format(format, args);
+    }
+
+    private void printCustom(String command, int commandIndent, String comment, int commentIndent) {
+        if (!command.isEmpty()) {
+            char[] indentChars = new char[commandIndent];
+            Arrays.fill(indentChars, ' ');
+            output.print(indentChars);
+
+            output.print(command);
+        }
+
+        if (!comment.isEmpty()) {
+            if (!command.isEmpty()) {
+                char[] indentChars = new char[Math.max(0, commentIndent - command.length() - commandIndent)];
+                Arrays.fill(indentChars, ' ');
+                output.print(indentChars);
+            }
+
+            output.print(comment);
+        }
+
+        output.println();
+    }
+
+    private void print(String command) {
+        printCustom(command, DEFAULT_COMMAND_INDENT, "", DEFAULT_COMMENT_INDENT);
+    }
+
+    private void print(String command, String comment) {
+        printCustom(command, DEFAULT_COMMAND_INDENT, comment, DEFAULT_COMMENT_INDENT);
+    }
+
+    private void printLabel(String label) {
+        printCustom(label, 0, "", 0);
+    }
+
+    private String[] split(String line) {
+        String[] split = line.split(COMMENT_START_CHAR);
+        if (split.length == 1) {
+            return new String[] {
+                split[0],
+                ""
+            };
+        } else {
+            String comment = COMMENT_START_CHAR + Arrays.stream(split).skip(1).collect(Collectors.joining(COMMENT_START_CHAR));
+
+            return new String[] {
+                split[0].trim(),
+                comment.trim()
+            };
+        }
+    }
+
+    @Override
+    public void writeAssembly(Iterable<FunctionInstructions> functions, OutputStream writer) {
+        output = new PrintStream(writer);
+
+        writeFileProlog();
+        for (FunctionInstructions function : functions) {
+            writeFunction(function);
+        }
+        writeFileEpilog();
+    }
+
+    private void writeFileProlog() {
+        print(".text");
+    }
+
+    private void writeFileEpilog() {
+        return;
+    }
+
+    private void writeFunction(FunctionInstructions function) {
+        writeFunctionProlog(function.getLdName());
+
+        for (String line : function.getInstructions()) {
+            String[] commandComment = split(line);
+            String command = commandComment[0];
+            String comment = commandComment[1];
+
+            if (command.equals(ApplyAssignment.FINAL_BLOCK_LABEL)) {
```

Comment:
`FINAL_BLOCK_LABEL` is only a substring of the command. Currently, the register allocator outputs `FINAL_BLOCK_LABEL + ":"` for the label itself and `"jmp " + FINAL_BLOCK_LABEL` for jumps